### ISSUE TITLE
Add "Delete via point" button to "Individual Route" => "Sort"

### ISSUE THREAD

**Issue by MagpieFourtyTwo** [cgeo/cgeo#8670]
Would be great to be able to [Trash] a via point from the list of via points in "Individual Route" => "Sort".

**Comment by moving-bits** [comment]
Yeah, I'm sort of longing for that myself. I've only been reluctant so far as to where to put that third button. If I put it in the same row as the sorting buttons, screen space for name etc. would get reduced significantly. Not sure, if that would work for all devices. (Other than that probably the easiest to use solution.) 
OTOH I would not like to open a second row of buttons for that, as this would produce a lot of empty space and require much more scrolling.
What do you think? / What would be your favorite solution?

**Comment by MagpieFourtyTwo** [comment]
What about just swiping it away (system conform to the left) ...?
Possibly there even can appear a trash can like with Android's notifications?

**Comment by moving-bits** [comment]
Closed with https://github.com/cgeo/cgeo/commit/fd0728343ceaaee7ecfc5288ae540df52f9e4878

**Comment by MagpieFourtyTwo** [comment]
Well done, works ... although I did not expect that I would have to additionally tap the trash bin.
Android 10's "delete notification swype" is different: You (really) swype the notification to the left, it in fact goes to the left as you do, then a trash appears - if you stop now, you can tap the can, but if you simply continue to swype, it will vanish immediately.
But, this is cosmetic, the real problem is:  It works only if you manage to do the horizontal swype *extremely* exact ...
And to be honest: I get it working only once out of ten.  :/
(BTW; I even have problems sometimes, to get the select mode activated by swyping to the right in a cache list)

So, I would prefer if the left swype will be accepted no matter how exact it is,
In fact a "grab" of the entry => If it goes to the left => delete and, if possible: If it goes up or down, this may be a quick sort move ...?  ;)

**Comment by moving-bits** [comment]
Right now I have implemented the same "swipe" behavior as we have in cache list - having no animation, but enabling/disabling some additional UI element (select box there, delete button here). Without animation it's hard to recognize that you're going to do a significant operation like deleting an element, therefore I decided to go with the delete button first. (There's no built-it swipe functionality (at least AFAIK), so you either have to add another library for that, or to hard-code it (which I would like to avoid).)

Regarding the "have to manage to do the horizontal swipe extremely exact": This is something we can help with :-) as we check the vertical deviation. I could set this to be a bit more tolerant.

**Comment by MagpieFourtyTwo** [comment]
I would highly appreciate that!  :)
And if possible, same same in cache list ... ;)
BTW: Why not set the derivation to half of the row height ...?

**Comment by moving-bits** [comment]
Hm, I need to correct my optimistic message above: Seems Android does not detect a fling gesture with "too much" deviation. We have an internal deviation limit, and I have increased it by factor three for testing purposes, but the `onFLing` method does not get called at all if I scroll/fling "too much diagonally". And when it's being called, values are way below our internal limit.

So, sadly, I cannot change this behavior.

What you can try is to make a shorter horizontal movement (as it's a fling gesture), such decreasing the possibility of too much vertical deviation.

**Comment by MagpieFourtyTwo** [comment]
Hmm. Bad news. Any chance to get an alternative way?
Cause with this limited swype you will not get it esp. on high res displays.
Tested again, but even with the shortest swype imaginable it's still just 1 out of 5.
What about a long tap ...?  Or a menu or button to activate delete-mode ...?

**Comment by moving-bits** [comment]
Long tap would be fine IMHO.

**Comment by MagpieFourtyTwo** [comment]
So what's about activating "Select mode" and "Delete mode" by long tap?
Although I would really like a possibility to sort vias of individual routes by Drag & Drop ...
So, just in case Drag & Drop would be a possibility, I would suggest
- Drag vertically: Sort
- Drag horizontally: Delete

...?

**Comment by MagpieFourtyTwo** [comment]
Ooops ... already implemented ... sorry ... but perhaps you can nevertheless take this into consideration ...?  ;)

**Comment by moving-bits** [comment]
> Although I would really like a possibility to sort vias of individual routes by Drag & Drop ...

I created a new issue for this for easier tracking.

**Comment by MagpieFourtyTwo** [comment]
Just tested the ``Long tap to delete`` and was in fact a little bit astonished that the tapped entry immediately disappeared ...  ;)
But then realized that there still is the ``Save`` button which can safely be ignored in order to get status quo antes. :)
Having this in mind I would say: Works. At least for now. Cause I'm really looking forward to the hopefully upcoming Drag'n Drop Sort. ;)

Nevertheless, I guess, this sort feature will mostly be used on cache series (like I made my test with).
Unfortunately, a lot of these have looong names and, even worse, with the significant part of the name at the very end (in form of a counter), which makes it really hard to say, which one is which ... so what would you think about a line wrap for cache names that do not fit into the column?  This, BTW in other cache lists as well?

And, yes, I know, you can use "screen rotate" as workaround to see long names in full, too, but to be honest: When being in the wild and navigating by map through a jungle, "screen rotate" is extremely annoying, sometimes even forces c:geo to crash, so most cachers I know have it disabled anyway ...

P.S.: While I do not immediately create an issue for every single idea I have, I do not want to be ignorant, would (and will) create one anyway, if it makes sense. but with questions like the one above I just want to clarify in advance, what others (and esp. those, who are implementing) think about it. ;)

**Comment by moving-bits** [comment]
For

> Unfortunately, a lot of these have looong names and, even worse, with the significant part of the name at the very end (in form of a counter), which makes it really hard to say, which one is which ... so what would you think about a line wrap for cache names that do not fit into the column?

see #8737
